### PR TITLE
Synchronise the express payment settings between the Cart & Checkout blocks

### LINKED ISSUE #50686: [Enhancement]: Synchronise express payment controls between Cart & Checkout blocks
### Describe the solution you'd like

In https://github.com/woocommerce/woocommerce/pull/47899, some controls for the express payment area in the Checkout block were introduced. This area (or inner block) is available in both the Cart & Checkout blocks, yet the controls only display for the Checkout block.

We need to display the same controls in the Cart block and find a way to synchronise them with the Checkout block. Changing controls and saving in one should reflect in the other.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by alexflorisca**
### Changes proposed in this Pull Request:

In #47899 I introduced some controls for the express payment inner block of the Checkout block. This is also displayed on the Cart block, but up until now the Cart block did not include these controls. This PR syncs the controls for the express payment area between the Cart and Checkout blocks. Saving in one, changes should be reflected in the other.

There's also a couple of minor visual tweaks in this PR, because they touch on the same files and it would conflict to raise different PRs. 

1) On the Cart block, if there are express payment buttons that have rounded corners, the bottom would be cut off because of the button padding.

| Before                                                                                                                                          | After                                                                                                                                           |
|-------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------|
| Screenshot 2024-08-14 at 15 43 49 | Screenshot 2024-08-14 at 15 42 33 |

2) Some text changes based on Nikki's feedback from this figma file (c8VYjXqluE9vlVSE1RgJix-fi-867_2724). Previously, there would be a notice displayed conditionally based on the toggle. I've removed this, and the paragraph above the button settings, in favour of some helper text for the "Apply uniform styles" toggle

| Before                                                                                                                                          | After                                                                                                                                           |
|-------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------|
| Screenshot 2024-08-14 at 17 21 58 | Screenshot 2024-08-14 at 17 16 01 |

Closes #50686.

### How to test the changes in this Pull Request:

**Prerequisites**

1. You will need to install the following plugin zips as they contain parallel work. [WooCommerce Stripe Gateway](https://github.com/user-attachments/files/16615462/woocommerce-gateway-stripe.zip) and [WooCommerce Payments](https://github.com/user-attachments/files/16615421/woocommerce-payments.zip) plugins
2. Install the [WooCommerce Payments Dev Tools plugin](https://github.com/Automattic/woocommerce-payments-dev-tools).
3. Check the `Force the platform_checkout_eligible flag (aka WooPay eligible) in the account cache` box
4. Check the `Enable the WooPay Express Checkout button` box

**Testing**
1. Make sure you have at least 2 express payment methods show up on the Checkout block
2. In two separate tabs, open up the pages with the Checkout and the Cart blocks in the editor.
3. On the Checkout block, click on the express payment block and change the button height and border radius. Save the post.
4. Refresh the page with the Cart block. You should see the new settings applied, both in the sidebar, and in the visual editor.
5. Toggle "Express Button Styles" to off and save the post.
6. Refresh the page with the Checkout block. 
7. Changes should be reflected here
8. Check the Cart & Checkout blocks on the front-end. Make sure the settings are respected in both.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Synchronise the express payment controls between the Cart & Checkout blocks

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/block-settings/index.tsx
Comment (mikejolley) [inline plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/block-settings/index.tsx]: I know this is all caps due to styling but these should use sentence case.

File: plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/block-settings/index.tsx
Comment (mikejolley) [inline plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/block-settings/index.tsx]: Sentence case here.

File: plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/block-settings/index.tsx
Comment (mikejolley) [inline plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/block-settings/index.tsx]: This is missing localisation.

File: plugins/woocommerce/src/Blocks/BlockTypes/CheckoutExpressPaymentBlock.php
Comment (mikejolley) [inline plugins/woocommerce/src/Blocks/BlockTypes/CheckoutExpressPaymentBlock.php]: Indentation seems off here.

File: plugins/woocommerce/src/Blocks/Utils/CartCheckoutUtils.php
Comment (mikejolley) [inline plugins/woocommerce/src/Blocks/Utils/CartCheckoutUtils.php]:
```suggestion
		if ( ! in_array( $cart_or_checkout, array( 'cart', 'checkout' ), true ) ) {
```

Would this simplify?

**Review by mikejolley — CHANGES_REQUESTED** [review]
Thanks @alexflorisca I left feedback for some missing parts.

The synchronisation appears to work between the cart and checkout blocks, but I didn't get any styling changes on the frontend. This was with 20px rounded borders small size (which worked in editor):

Screenshot 2024-08-15 at 16 27 29
Screenshot 2024-08-15 at 16 27 05

Also note the google pay button is cut off on the left and right sides in the editor which can be seen in the screenshot in the PR description.

**Comment by alexflorisca** [comment]
Thanks for the review Mike. I've made all the changes you mentioned inline.

> The synchronisation appears to work between the cart and checkout blocks, but I didn't get any styling changes on the frontend. This was with 20px rounded borders small size (which worked in editor):

So the while the values are passed down to the various express buttons, currently only the WooPay buttons respects the boderRadius setting. WooPay have also done some work in parallel with this to add a new Stripe API which supports borderRadius, but I'm not sure it's live yet. It's on my radar to make sure this is included in their work. More info in [this comment ](https://github.com/woocommerce/woocommerce/pull/47899#issuecomment-2288310568)

> Also note the google pay button is cut off on the left and right sides in the editor which can be seen in the screenshot in the PR description.

This is because images are rendered in the editor instead of the buttons. This isn't perfect currently and there might be situations where some buttons are slightly cut off, but it's an interim solution (and better than what we had) before we start rendering the buttons in the editor

File: plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/block-settings/index.tsx
Comment (mikejolley) [inline plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/block-settings/index.tsx]: Panels do use title case I think

**Review by mikejolley — APPROVED** [review]
@alexflorisca I will pre-approve this since the things I experienced seem to be expected and the other functionality worked fine. But before merging please:

1. Check that panel title. I think panels use title case, and settings within panels use sentence case. 
2. Fix CI and linting errors. See https://github.com/woocommerce/woocommerce/actions/runs/10408846819/job/28827165337?pr=50688

Thanks!

**Comment by senadir** [comment]
@alexflorisca should be in draft for now?

**Comment by alexflorisca** [comment]
I will merge this into a feature branch once I merge the revert

**Comment by alexflorisca** [comment]
For context, anyone coming across this, this PR is part 2 of 2. The first PR is https://github.com/woocommerce/woocommerce/pull/47899. This was reverted in https://github.com/woocommerce/woocommerce/pull/50763. It was decided to create a feature branch and wait one release for more/parallel work to be completed in order to ship a more stable experience for merchants. This will now merge into `feature/express-payment-improvements`

**Pull request merged**
